exchanges/graphcache/package.json

```diff
@@ -63,10 +63,16 @@
     "prepublishOnly": "run-s clean build"
   },
   "peerDependencies": {
-    "@urql/core": "^5.0.0"
+    "@urql/core": "^5.0.0",
+    "graphql": "^14.0.0 || ^15.0.0 || ^16.0.0"
```

Comment:
Hm, I understand that Yarn PnP may have problems here, but since this is a typings change, I'm not sure it'll actually be of huge concern, especially since there's overlap here with what `graphql.web` does.

I'm a little concerned that this change would cause other package managers (or also Yarn Berry) to install `graphql` automatically. The problem is that peers have changed in how they're treated by package managers a lot recently, and auto-installations are now a thing, and this might now have the unintended side-effect of always installing `graphql`, even if it's marked as optional.

I think it'd be nice to split this out and only port the typings change for now. Happy to go ahead and make that change, since I dropped the ball on getting the main typings fix in first

Reply:
> I think it'd be nice to split this out and only port the typings change for now. Happy to go ahead and make that change, since I dropped the ball on getting the main typings fix in first

That makes sense. You can go ahead and close this PR then.

I'll test the inference again after the typing fix is released. If it's still not working, though as you said there should be sufficient overlap, then I'll open another PR with the peer dependency change. My understanding is that if it's also an optional one package managers won't auto install it, but I'll need to do some digging to confirm that.